src/hotspot/share/runtime/mutex.cpp

```diff
@@ -297,6 +297,7 @@ Mutex::Mutex(int Rank, const char * name, SafepointCheckRequired safepoint_check
          "Safepoint check never locks should always allow the vm to block: %s", name);
 
   assert(_rank >= 0, "Bad lock rank: %s", name);
+  assert(_rank <= nonleaf, "Bad lock rank: %s", name);
```

Comment:
Can we just combine these into a single range check? And really the assert message should include the bad rank value. And probably this basic range check should be first.

Reply:
I combined the check and moved it up for now.  In my next change that makes Rank into an enum class, the range check is different because you can't add to nonleaf which is going to be renamed 'safepoint' rank.